src/main/java/com/aws/greengrass/logmanager/model/ProcessingFileLRU.java

```diff
@@ -0,0 +1,109 @@
+package com.aws.greengrass.logmanager.model;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+
+/**
+ * LRU cache that holds information about each file that is being processed. As long as a log file is in the file system
+ * and can be found, we can't know for sure if that file won't get new contents written to it in the future.
+ */
+public class ProcessingFileLRU {
```

Comment:
Still need to handle locking here. Although we only have 1 thread

Reply:
Did you look at using the built-in LinkedHashMap?

Here's an example of how we used to use it in CDA: https://github.com/aws-greengrass/aws-greengrass-client-device-auth/commit/982fe21e7b8171bd01f80e92458600482769c983#diff-dbfa22ec04fcd9b53a287244e4b4ce88dbd005eafefe4f3ba69e371812d3b552R28

Reply:
Nope ahaha I will update to use that